rtrlib/rtr/new-api.h

```diff
@@ -0,0 +1,136 @@
+//Internal types
+struct rtr_transport_socket;
+struct rtr_mgr_group;
+struct rtr_mgr;
+
+//external types
+typedef struct rtr_transport_socket rtr_transport_socket_t;
+typedef struct rtr_mgr_group rtr_mgr_group_t;
+typedef struct rtr_mgr rtr_mgr_t;
+
```

Comment:
1. We decided to use the kernel coding style. The style guide says:

> 5) Typedefs
> -----------
> 
> Please don't use things like ``vps_t``.
> It's a **mistake** to use typedef for structures and pointers. When you see a
> 
> .. code-block:: c
> 
> 
> 	vps_t a;
> 
> in the source, what does it mean?
> In contrast, if it says
> 
> .. code-block:: c
> 
> 	struct virtual_container *a;
> 
> you can actually tell what ``a`` is.
https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/plain/Documentation/process/coding-style.rst

2. The POSIX standard forbids to use _t for variable names:
https://stackoverflow.com/a/12727104/537958